packages/wrangler/src/config/validation.ts

```diff
@@ -1026,41 +1034,9 @@ const validateTailConsumers: ValidatorFn = (diagnostics, field, value) => {
 	return isValid;
 };
 
-/**
- * Validate top-level environment configuration and return the normalized values.
- */
-function normalizeAndValidateEnvironment(
-	diagnostics: Diagnostics,
-	configPath: string | undefined,
-	topLevelEnv: RawEnvironment,
-	isDispatchNamespace: boolean
-): Environment;
 /**
  * Validate the named environment configuration and return the normalized values.
  */
-function normalizeAndValidateEnvironment(
-	diagnostics: Diagnostics,
-	configPath: string | undefined,
-	rawEnv: RawEnvironment,
-	isDispatchNamespace: boolean,
-	envName: string,
```

Comment:
The idea of this overload was that if you specified an `envName` then you "had" to also specify all the other optional params.